File: src/main/java/com/garageos/modules/estimate/service/EstimateService.java
package com.garageos.modules.estimate.service;

import com.garageos.modules.estimate.dto.request.CreateEstimateRequest;
import com.garageos.modules.estimate.dto.response.EstimateResponse;

import java.util.List;

public interface EstimateService {

    EstimateResponse createEstimate(
            CreateEstimateRequest request);

    EstimateResponse getEstimate(Long id);

    List<EstimateResponse> getAllEstimates();

    EstimateResponse updateEstimate(
            Long id,
            CreateEstimateRequest request);

    void deleteEstimate(Long id);
    EstimateResponse approveEstimate(Long id);

    EstimateResponse rejectEstimate(Long id);
    EstimateResponse createEstimate(String jobCardNumber);
    EstimateResponse approveEstimate(String jobCardNumber);
}

File: src/main/resources/static/services/estimateService.js
window.EstimateService = {

    async createEstimate(request) {

        return await Api.post(

            "/estimates",

            request

        );

    },

    async addEstimateItem(
        estimateId,
        request
    ) {

        return await Api.post(

            `/estimates/${estimateId}/items`,

            request

        );

    },

    async finishEstimate(estimateId, payload) {

        return await Api.put(
            `/estimates/${estimateId}`,
            payload
        );

    },

    async approveEstimate(jobCardNumber) {

        return await Api.post(
            `/workflow/${jobCardNumber}/estimate/approve`
        );

    },
    async updateEstimate(id, request) {

        return await Api.put(
            `/estimates/${id}`,
            request
        );

    },

//    async getEstimate(id) {
//        return await Api.get(`/estimates/${id}`);
//    }
//
//    async rejectEstimate(id) {
//        return await Api.put(`/estimates/${id}/reject`);
//    }
};